### DISCUSSION TITLE
Search string for empty date

### DISCUSSION THREAD

**Discussion by nthoman** [microsoft-search/pnp-modern-search#3626] [Q&A]
`$SiteURL = "https://{Yours}.sharepoint.com/sites/Reports"
$SearchQuery = "ReportRun:1900-01-01 Path:" + $SiteURL

Connect-PnPOnline -Url $SiteURL -Interactive

Write-Host "Searching for: $SearchQuery" -ForegroundColor Cyan
$SearchResults = Submit-PnPSearchQuery -Query $SearchQuery -All -SortList @{Created="Descending"}`

I am using the Powershell code above to try and find all the files that don't have ReportRun defined. However it's not returning anything. ReportRun is the correct field name and it is Indexed. What's the correct way to structure the Search Query? I also mapped the ReportRun field to "RefinableDate01". If that helps.

**Comment by kasperbolarsen** [comment]
Have you checked that value you are looking for? You might use one of the tools here, https://microsoft-search.github.io/pnp-modern-search/using-query-tools-to-verify-issues/
To inspect the contents returned by the API

**Comment by wobba** [comment]
See if my old post helps you https://www.techmikael.com/2014/08/how-to-search-up-items-which-dont-have.html if using date properties. If not map them to text properties as well and see  https://www.techmikael.com/2023/04/filter-on-managed-properties-in-search.html

**Comment by xXMrNidaXx** [comment]
Great question on benchmarking! Here are some pointers from our experience:

**Key metrics to track:**
- Time-to-first-token (TTFT) - crucial for user experience
- Tokens/sec throughput
- Retrieval latency vs. generation latency (break these down separately)

**Performance tip:** If you're seeing slow TTFT, look into caching strategies. There's a recent paper called "RAGCache" that shows 4x TTFT improvement by caching intermediate embedding states.

Happy to share more details if helpful!
